File: Practica General/POO/Ejercicio 3/Coche.java
public class Coche {
    private String marca;
    private String modelo;
    private int velocidad;

    public Coche(String marca, String modelo) {
        this.marca = marca;
        this.modelo = modelo;
        this.velocidad = 0;
    }

    public void acelerar() {
        this.velocidad += 10;
    }

    public void frenar() {
        this.velocidad = Math.max(0, this.velocidad - 5);
    }

    public void mostrarVelocidad() {
        System.out.println(this.marca + " " + this.modelo + " - Velocidad: " + this.velocidad + " km/h");
    }

    public static void main(String[] args) {
        Coche c1 = new Coche("Toyota", "Corolla");
        Coche c2 = new Coche("Ford", "Fiesta");

        // Acelerar
        c1.acelerar();
        c1.acelerar();
        c2.acelerar();

        // Frenar
        c1.frenar();
        c2.frenar();
        c2.frenar();

        // Mostrar velocidad
        c1.mostrarVelocidad();
        c2.mostrarVelocidad();
    }
}

File: Practica General/POO/Ejercicio 3/coche.py
class Coche:
    def __init__(self, marca, modelo):
        self.marca = marca
        self.modelo = modelo
        self.velocidad = 0

    def acelerar(self):
        self.velocidad += 10

    def frenar(self):
        self.velocidad = max(0, self.velocidad - 5)

    def mostrar_velocidad(self):
        print(f"{self.marca} {self.modelo} - Velocidad: {self.velocidad} km/h")


# Prueba
c1 = Coche("Toyota", "Corolla")
c2 = Coche("Ford", "Fiesta")

# Acelerar
c1.acelerar()
c1.acelerar()
c2.acelerar()

# Frenar
c1.frenar()
c2.frenar()
c2.frenar()

# Mostrar velocidad
c1.mostrar_velocidad()
c2.mostrar_velocidad()
